# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/nsCaret.h
# Commit: b32811ea1c61
# Full Hash: b32811ea1c61e1891454d826f7ba003416c9eb57
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2024-03-11 21:13:39
# Regressor Bug: 1860328
# File Overlap Count: 2
# Description:
#   Bug 1860328 - Fix a crash which a caret frame misses a MarkNeedsDisplayItemRebuild() call r=emilio
#   
#   Generally, we mark a caret frame for display first, and then
#   nsCaret tracks this frame in nsCaret::SchedulePaint to call
#   MarkNeedsDisplayItemRebuild() accordingly. However, it's possible
# ==============================================================================

diff -r caab023e5d2d -r b32811ea1c61 layout/base/nsCaret.h
--- a/layout/base/nsCaret.h	Mon Mar 11 13:41:27 2024 +0000
+++ b/layout/base/nsCaret.h	Mon Mar 11 13:41:28 2024 +0000
@@ -197,6 +197,13 @@
 
   size_t SizeOfIncludingThis(mozilla::MallocSizeOf aMallocSizeOf) const;
 
+  void SetLastCaretFrame(nsIFrame* aLastCaretFrame) {
+    if (mLastCaretFrame && mLastCaretFrame != aLastCaretFrame) {
+      mLastCaretFrame->MarkNeedsDisplayItemRebuild();
+    }
+    mLastCaretFrame = aLastCaretFrame;
+  }
+
  protected:
   static void CaretBlinkCallback(nsITimer* aTimer, void* aClosure);
 